atp/server.go

```diff
@@ -91,11 +100,49 @@ func RunATPServer( //nolint:funlen
 			"",
 		}); err != nil && !closed {
 			goroutineError = fmt.Errorf("failed to encode CBOR response (%w)", err)
-			panic(goroutineError)
+			errch <- goroutineError
+			return
 		}
 	}()
 
 	// Keep running until both goroutines are done
 	wg.Wait()
-	return goroutineError
+	return errchError
+}
+
+// ServerChannel holds the methods to talking to an ATP server (plugin).
+type ServerChannel interface {
+	io.Reader
+	io.Writer
+	io.Closer
+}
+
+type ATPServer struct {
+	channel ServerChannel
+	decMode cbor.DecMode
+	logger  log.Logger
+	Decoder *cbor.Decoder
+	Encoder *cbor.Encoder
+}
+
+func NewATPServer(
```

Comment:
Is this only for the test cases at the moment?

Reply:
Yep